src/node.rs

```diff
@@ -491,6 +595,12 @@ impl<Lifetime, K, V, Mutability, NodeType, HandleType> Handle<NodeRef<Lifetime,
     }
 }
 
+impl<Lifetime, K, V, NodeType, HandleType> Handle<NodeRef<Lifetime, K, V, marker::Mut, NodeType>, HandleType> {
+    pub fn reborrow_mut(&mut self) -> Handle<NodeRef<marker::Borrowed, K, V, marker::Mut, NodeType>, HandleType> {
```

Comment:
Unfortunately, this function is actually unsafe. For example, something like `node.reborrow_mut().ascend().unwrap().pop()` could leave `node` a dangling pointer.

I can think of a few things that would make this safe, like disallowing `ascend` on the returned node, but any such solution would require even more type parameters.